packages/patterns/src/types.js

```diff
@@ -526,11 +526,17 @@ export {};
 /**
  * @template {Record<string | symbol, MethodGuard>} [T=Record<string | symbol, MethodGuard>]
  * @typedef {{
- *   klass: 'Interface',
  *   interfaceName: string,
  *   methodGuards: { [K in keyof T]: K extends symbol ? never : T[K] },
  *   symbolMethodGuards?: CopyMap<(keyof T) & symbol, MethodGuard>,
  *   sloppy?: boolean,
+ * }} InterfaceGuardPayload
+ */
+
+/**
+ * @template {Record<string | symbol, MethodGuard>} [T=Record<string | symbol, MethodGuard>]
+ * @typedef {{
+ *   klass: 'Interface' & InterfaceGuardPayload<T>,
  * }} InterfaceGuard
```

Comment:
I'm pretty sure this applies InterfaceGuardPayload\<T> to the value of the `klass` property, and am surprised it didn't cause errors: [TypeScript Playground](https://www.typescriptlang.org/play?filetype=js&useJavaScript=truee=4#code/PQKhCgAIUgBAXApgWwA4BsCGTIG8DO8ATgJYB2A5pAD6T4CeyARgPboC+kA2gNIC8ACkKlKNOo1boAlAF0oMBCgzZEeAK5kA1mRYB3Mpy4A1Phu16yc6HHj1UiACaIAZngBKiAMYsiDgDw8ADSQRgB8nADCLKj0ALKYqPLA4OCgENaKaFg4uGY6+oaxiPAAFiwOAOJqmL6mWvmW8jZK2aq4Ht6+fsLkVLQMzGzBRaXlVTUO4dwAKnwdPv49ov0SQ5AjZZXVvqFWCrb2Tq64uE2QkORIRM6YnogAcpjIiABcdMS9gWeQz6NbE-g3rhuDwLmRIJpEPQWK5pjI3qDEAAPJBkBz4cSDdCQAD8kDIiAAbogiJA3tNeDJIOwvtZzgNJBsxtt0Ti3lEYvFUH4BJDobCpJAAGSYyTDYqbcY7WkwenoaIxNmQVhsRCYMgy6mcACSZCuNzuUocAAVMPR5ZgHEkUmkmpllDk8hZChLmRM6uZ9HtmlkVO4vAtuh9lqK1kz-jtDLN5l0ln1Q+hxX8jbs7QdHC48Kc6RCsPhAZAAOS6-W3RCF4WQEskg2II2m80sS1+aahTXsHV6mtlo0ARmtqTAaZafty9Wd3HDRo9DW99ta-s6i2D8YZYddEcmUbmAdjK7Ea8T6w3Kbn6aOWc0eYLxa71zLhc4Iur98NLIbFv8rc7pbfEwATAOaQ2HYqgvrWfbQMk3hkIQkCoEQLBMOgKCQHweC5pg+ZvAAROBZY4cElzdncjzPLhABiLAsIRPwniyBa4Jw7AANw2mAIH2FWd4QSy-5QeAMFwUwloAPrEa+DxPKo6HAleWEFnhPEEURymkdJbwAAzBL8koMUCzFsYO+ygdxv51nxAlCfAkAUNRDhoRh8nYZASnmbREm1mRryuVRNE6fRAIGdSLFAA)
```suggestion
 * @typedef {{klass: 'Interface'} & InterfaceGuardPayload<T>} InterfaceGuard
```